crates/data_components/src/databricks/sql_warehouse.rs

```diff
@@ -0,0 +1,568 @@
+        let links = response
+            .get_mut("result")
+            .and_then(|result| result.get_mut("external_links").map(Value::take))
+            .ok_or_else(|| {
+                MissingJsonFieldSnafu {
+                    field: "result.external_links",
+                }
+                .build()
+            })?;
+
+        let Value::Array(links) = links else {
+            return Err(Error::InvalidJsonArray {
+                field: "external_links".into(),
+            });
+        };
+
+        links
+            .into_iter()
+            .map(|link| serde_json::from_value(link).context(DeserializeExternalLinkFailedSnafu))
+            .collect()
+    }
+
+    async fn fetch_chunk_data(&self, url: &str) -> Result<bytes::Bytes, Error> {
+        self.client
+            .get(url)
+            .send()
+            .await
+            .context(HttpRequestFailedSnafu)?
+            .error_for_status()
+            .context(HttpRequestFailedSnafu)?
+            .bytes()
+            .await
+            .context(HttpRequestFailedSnafu)
+    }
+
+    fn read_arrow_batches(
+        bytes: bytes::Bytes,
+    ) -> Result<Vec<arrow::record_batch::RecordBatch>, Error> {
+        let cursor = Cursor::new(bytes);
+        let reader = StreamReader::try_new(cursor, None).context(ArrowStreamReadFailedSnafu)?;
+        Ok(reader
+            .collect::<Result<Vec<_>, _>>()
+            .context(ArrowStreamReadFailedSnafu)?
+            .into_iter()
+            .filter(|batch| batch.num_rows() > 0)
+            .collect())
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ExternalLink {
+    chunk_index: u64,
+    external_link: String,
+}
+
+fn map_databricks_type(type_name: &str) -> Result<DataType, Error> {
+    if type_name.to_uppercase().starts_with("DECIMAL") {
+        let (precision, scale) = if type_name.contains('(') {
+            let params: &str = type_name
+                .split('(')
+                .nth(1)
+                .ok_or_else(|| Error::UnsupportedType {
+                    ty: type_name.to_string(),
+                })?
+                .trim_end_matches(')')
+                .trim();
+
+            let parts: Vec<&str> = params.split(',').map(str::trim).collect();
+            if parts.len() == 2 {
+                let precision: u8 = parts[0].parse().map_err(|_| Error::UnsupportedType {
+                    ty: type_name.to_string(),
+                })?;
+                let scale: i8 = parts[1].parse().map_err(|_| Error::UnsupportedType {
+                    ty: type_name.to_string(),
+                })?;
+                (precision, scale)
+            } else {
+                return Err(Error::UnsupportedType {
+                    ty: type_name.to_string(),
+                });
+            }
+        } else {
+            (10, 0)
+        };
+
+        return Ok(DataType::Decimal128(precision, scale));
+    }
```

Comment:
```suggestion
    if type_name.to_uppercase().starts_with("DECIMAL") {
        let (precision, scale) = match type_name.split_once('(') {
            Some((_, params)) => {
                let params = params.trim_end_matches(')').trim();
                let parts: Vec<_> = params.split(',').map(str::trim).collect();
                if parts.len() != 2 {
                    return Err(Error::UnsupportedType { ty: type_name.to_string() });
                }
                let precision: u8 = parts[0].parse().map_err(|_| Error::UnsupportedType {
                    ty: type_name.to_string(),
                })?;
                let scale: i8 = parts[1].parse().map_err(|_| Error::UnsupportedType {
                    ty: type_name.to_string(),
                })?;
                (precision, scale)
            }
            None => (10, 0),
        };
        return Ok(DataType::Decimal128(precision, scale))
    }
```

I think this is slightly cleaner.